### DISCUSSION TITLE
Extracting thumbnail from a JPG file

### DISCUSSION THREAD

**Discussion by lkintact** [ImageMagick/ImageMagick#7661] [Help]
### ImageMagick version

7.1.1-38 Q16-HDRI x64 b0ab922:20240901

### Operating system, version and so on

Windows 10 Home

### Description

Hi again, everyone,

I wanted to extract an EXIF thumbnail from a [JPG image](https://github.com/user-attachments/assets/4fd5eda0-26c4-4eea-837c-55da1a6f483b), for which I tried this command:
`magick.exe image.jpg thumbnail:thumb.jpg` (that's a solution I found on the [ImageMagick forum](https://imagemagick.org/discourse-server/viewtopic.php?t=15853)).

That didn't work, IM reported this message:
`magick.exe: image does not have a EXIF thumbnail ``thumb.jpg' @ error/thumbnail.c/WriteTHUMBNAILImage/190.` 

`magick.exe identify -verbose image.jpg`, however, shows that the image does have a thumbnail.

Please tell me a way to extract it.

**Comment by snibgo** [comment]
IM and exiftool both say the file has a thumbnail. I suppose there is a bug in coders/thumbnail.c

As a workaround, exiftool can extract the thumbnail:
```
exiftool -ThumbnailImage -b flowerPerson.jpg >x.jpg
```
... where `flowerPerson.jpg` is my name for your file.

**Comment by lkintact** [comment]
@snibgo: Thank you!

> I suppose there is a bug in coders/thumbnail.c

Shall I open a new issue about this?

Edit: As extracting thumbnails doesn't seem to work with other images either (see my message below), I decided to create an issue.

**Comment by lkintact** [comment]
I tried [another JPEG image](https://github.com/user-attachments/assets/beeb3544-7680-4de8-b102-8db65c199679) with a thumbnail, and, again, `magick.exe image.jpg thumbnail:thumb.jpg` didn't extract it, returning the same error message.
